go-runtime/encoding/encoding.go

```diff
@@ -278,6 +277,59 @@ func decodeValue(d *json.Decoder, v reflect.Value) error {
 	}
 }
 
+// isNextTokenNull implements a cheap/dirty version of `Peek()`, which json.Decoder does
+// not support. This function is used to conditionally execute logic for handling nulls
+// without popping the next token, so that downstream code for the non-null case is still
+// able to pop it. Remember to pop the next token (d.Token()) inside your conditional
```

Comment:
nit: i wonder if we should also handle consuming this token in `isNextTokenNull` (and maybe rename/change signature as needed) so callers don't need to manage this

i don't feel strongly!

Reply:
ooooo callbacks! Done! Looks way better now :D